### ISSUE TITLE
Codegen failure "Assert failed '!type->IsNonVectorComposite()'"

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1312]
Still using the same project from https://github.com/beefytech/Beef/issues/1302, with the changes from https://github.com/beefytech/Beef/issues/1311. If you change the code of the property `Position` of the class `SDLWindow` to the code below, the codegen failure mentioned in the issue title happens.
```
	    public Point Position
	    {
	        get => position;
	        set
	        {
	            position = value;
	            ScheduleCommand(() => SDL.SetWindowPosition(SDLWindowHandle, value.X, value.Y));

	    	    ScheduleEvent(=> PositionChanged); // RIP Codegen
	        }
	    }

	    public Event<Action> PositionChanged ~ _.Dispose();
```

**Comment by bfiete** [comment]
Clearly not a lot of people have been using method references!

**Comment by bfiete** [comment]
Fixed at fd8e2dd2327a6e48b015d07995f6ca4a11057d46
